part/map.go

```diff
@@ -243,9 +243,11 @@ func (m *Map[K, V]) UnmarshalJSON(data []byte) error {
 
 func (m Map[K, V]) MarshalYAML() (any, error) {
 	kvs := make([]mapKVPair[K, V], 0, m.Len())
-	iter := m.tree.Iterator()
-	for _, kv, ok := iter.Next(); ok; _, kv, ok = iter.Next() {
-		kvs = append(kvs, kv)
+	if m.tree != nil {
+		iter := m.tree.Iterator()
+		for _, kv, ok := iter.Next(); ok; _, kv, ok = iter.Next() {
+			kvs = append(kvs, kv)
+		}
```

Comment:
```suggestion
	if m.tree == nil {
		return kvs, nil
	}

	iter := m.tree.Iterator()
	for _, kv, ok := iter.Next(); ok; _, kv, ok = iter.Next() {
		kvs = append(kvs, kv)
```
WDYT about flipping the if statement?

Reply:
I think in this case I prefer the way I wrote it: you start with empty and you populate it when there's values to put in. it's much shorter too. Though in general it's often better to do the early return.